specs/_features/ePBS/validator.md

```diff
@@ -0,0 +1,243 @@
+Proposer is introduced to construct and broadcast `InclusionList` alongside `SignedBeaconBlock`.
+- Proposer for slot `N` submits `SignedBeaconBlock` and in parallel submits `InclusionList` to be included at the beginning of slot `N+1`.
+- Within `InclusionList`, `Transactions` are list of transactions that the proposer wants to include at the beginning of slot `N+1`.
+- Within `inclusionList`, `Summaries` are lists consisting on addresses sending those transactions and their gas limits. The summaries are signed by the proposer.
+- Proposer is allowed to send many of these pairs that aren't committed to its beacon block so no double proposing slashing is involved.
+
+#### Constructing the `InclusionList`
+
+To obtain an inclusion, a block proposer building a block on top of a `state` must take the following actions:
+
+1. Retrieve inclusion list by calling `get_inclusion_list`.
+
+```python
+def get_inclusion_list(payload_id: Optional[PayloadId], execution_engine: ExecutionEngine) -> InclusionListResponse:
+        return execution_engine.get_payload(payload_id).inclusion_list
+```
+
+2. Build `InclusionListSummary`
+ - Set `summary` to the `inclusion_list_response.summary`
+ - Set `proposer_index` to the proposer's index
+
+3. Get `signature` by signing `InclusionListSummary` with its private key.
+
+```python
+def get_inclusion_list_summary_signature(state: BeaconState, inclusion_list_summary: InclusionListSummary, block_slot: Slot, privkey: int) -> BLSSignature:
+    domain = get_domain(state, DOMAIN_INCLUSION_LIST, compute_epoch_at_slot(block_slot))
+    signing_root = compute_signing_root(inclusion_list_summary, domain)
+    return bls.Sign(privkey, signing_root)
+```
+
+4. Build `SignedInclusionListSummary`
+ - set `signed_inclusion_list_summary.summary` to `InclusionListSummary`
+ - set `signed_inclusion_list_summary.signature` to `signature
+
+5. Build `InclusionList
+- set `inclusion_list.summaries` to `SignedInclusionListSummary`
+- set `inclusion_list.transactions` to `inclusion_list_response.transactions`
+
+#### Broadcast inclusion list
+
+Finally, the validator broadcasts inclusion to the inclusion list subnet, the `inclusion_list` pubsub topic.
+
+### Block proposal
+
+#### Constructing the new `signed_execution_payload_header_envelope` field in  `BeaconBlockBody`
+
+To obtain `signed_execution_payload_header_envelope`, a block proposer building a block on top of a `state` must take the following actions:
+* Listen to the `execution_payload_header` gossip subnet
+* Filter out the bids where `signed_execution_payload_header_envelope.message.header.parent_hash` matches `state.latest_execution_payload_header.block_hash`
+* The `signed_execution_payload_header_envelope` must satisfy the verification conditions found in `process_execution_payload_header`.
+* Select the best bid and set `body.signed_execution_payload_header_envelope = signed_execution_payload_header_envelope`
```

Comment:
Do we want to specify "best bid"  here? I think proposers should be able to select "a bid".